libevm/VM.cpp

```diff
@@ -1096,7 +1492,9 @@ void VM::interpretCases()
 #if EVM_HACK_DUP_64
 			*(uint64_t*)m_SPP = *(uint64_t*)(m_SP + n);
 #else
-			m_SPP[0] = m_SP[n];
+			// the stack slot being copied into may no longer hold a u256
+			// so we construct a new one in the memory, rather than assign
+			new(m_SPP) u256(m_SP[n]);
```

Comment:
Oh - we will need XDUP and XSWAP too.  Either that or implementations will be forced to keep track of the type of every stack slot, and this routine will involve checking the type of the source and destination.  In C++ we can fix that by making u256 a POD, but some other languages can't get that close to the metal.